WordPress/Classes/ViewRelated/Reader/ReaderInterestsCoordinator.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+
+class ReaderSelectInterestsCoordinator {
+    private struct Constants {
+        static let userDefaultsKeyFormat = "Reader.SelectInterests.hasSeenBefore.%@"
+    }
+
+    private let interestsService: ReaderFollowedInterestsService
+    private let store: KeyValueDatabase
+    private let userId: NSNumber?
+
+    /// Creates a new instance of the coordinator
+    /// - Parameter service: An Optional `ReaderFollowedInterestsService` to use. If this is `nil` one will be created on the main context
+    ///   - store: An optional backing store to keep track of if the user has seen the select interests view or not
+    ///   - userId: The logged in user account, this makes sure the tracking is a per-user basis
+    init(service: ReaderFollowedInterestsService? = nil,
+         store: KeyValueDatabase = UserDefaults.standard,
+         userId: NSNumber? = nil) {
+
+        let defaultContext = ContextManager.sharedInstance().mainContext
```

Comment:
I would avoid creating inline instances. If `ReaderSelectInterestsCoordinator` needs a context that should be a param in the constructor.

Reply:
Good catch, done in: 1b6a10258e1234026741cf11e6502bfb485ab949